### PR TITLE
fluent: mandate slug names to be prefixed by crate name

### PR DESCRIPTION
**Pull request by Xiretza**
This is currently only convention, but not actively checked for.

Additionally, improve error messages to highlight the path of the offending fluent file rather than the identifier preceding it.

This will conflict with #100671, so I'll leave it as draft until that's merged.

### CODE REVIEW

File: compiler/rustc_macros/src/diagnostics/fluent.rs [hunk 1]
```diff
@@ -205,11 +205,23 @@ pub(crate) fn fluent_messages(input: proc_macro::TokenStream) -> proc_macro::Tok
...
-                    &name.replace('-', "_").replace(&format!("{}_", res.ident), ""),
-                    span,
-                );
+                let crate_prefix = format!("{}_", res.ident);
+
+                let snake_name = name.replace('-', "_");
+                let snake_name = match snake_name.strip_prefix(&crate_prefix) {
+                    Some(rest) => Ident::new(rest, span),
+                    None => {
+                        Diagnostic::spanned(
```
Comment (davidtwco) [inline compiler/rustc_macros/src/diagnostics/fluent.rs:214]: This is great, could you add a failing test in `src/test/ui-fulldeps/fluent-messages/test.rs`?

Comment (davidtwco) [reply compiler/rustc_macros/src/diagnostics/fluent.rs:214]: I suspect this test will be failing anyway given the `Span` changes, so you would have to bless it anyway.

Comment (Xiretza) [reply compiler/rustc_macros/src/diagnostics/fluent.rs:214]: Yeah, had to update a couple tests, should be good now.

**Review by davidtwco — APPROVED** [review]
Great! r=me once #100671 lands

**Pull request merged**